File: Auth/Users.App/Interface/ISentEmailsRepo.cs
using Users.Domain;

namespace Users.App.Interface
{
    public interface ISentEmailsRepo
    {
        Task<IList<SentEmail>> GetSentEmailsForDateAsync(Guid userId, SentEmailType emailType, DateTimeOffset date);
        Task InsertAsync(SentEmail sentEmail);
    }
}

File: NodeAuth/src/repos/interfaces/ISentEmailsRepo.ts
import { SentEmail, SentEmailType } from "../../models/SentEmail";

export interface ISentEmailsRepo {
  getSentEmailsForDateAsync(
    userId: string,
    emailType: SentEmailType,
    date: Date,
  ): Promise<SentEmail[]>;
  insertAsync(sentEmail: SentEmail): Promise<void>;
}
